src/dotnet/Telemetry.cs

```diff
@@ -63,7 +90,18 @@ public void TrackEvent(string eventName, IDictionary<string, string> properties,
             {
                 return;
             }
+            if(!_shouldLogTelemetry)
+            {
+                return;
+            }                
 
+            _trackEventTask = Task.Factory.StartNew(
```

Comment:
The way this is implemented it means an instance of a TelemetryClient can only be used to track a single event.  Is that intended?